pkg/data/values.go

```diff
@@ -18,18 +28,51 @@ func GetValueN(data map[string]interface{}, keys ...string) interface{} {
 	return val
 }
 
+// GetValue works the same as GetValueFromAny. Kept this way to avoid breaking changes with the previous interface,
+// GetValueFromAny should be used directly in most cases.
 func GetValue(data map[string]interface{}, keys ...string) (interface{}, bool) {
+	return GetValueFromAny(data, keys...)
+}
+
+// GetValueFromAny retrieves a value from the provided collection, which must be a map[string]interface or a []interface.
+// Keys are always strings.
+// For a map, a key denotes the key in the map whose value we want to retrieve.
+// For the slice, it denotes the index (starting at 0) of the value we want to retrieve.
+// Returns the retrieved value (if any) and a bool indicating if the value was found.
+func GetValueFromAny(data interface{}, keys ...string) (interface{}, bool) {
```

Comment:
Wonder whether it would be a good idea to allows  `keys` to be `any` and validate that they're `string` or a number..
This way we could be sure that `-1` passed in expected a list vs `"-1"` which would expect a map.

This does add more complexity with this function like what to return if you receive an unexpected type, etc. So I think it's okay as is.

Reply:
I think that the way this function is used in Steve is that keys are essentially user provided input. So if we wen that direction Steve would need to do some selective parsing there, which I don't think is desirable. 

That being said, I think that you are correct that the underlying library really should be more specific than this. I tried to implement a more comprehensive approach that (IIRC) did what you suggested in #325 (though I doubt that will merge). I'd say if we want an improvement like that we should consider a rewrite like that Pr looks for.